dotCMS/src/integration-test/java/com/dotcms/datagen/AppDescriptorDataGen.java

```diff
@@ -0,0 +1,227 @@
+        basePath = Paths.get(basePath).normalize().toString();
+        final File file = new File(basePath, fileName);
+        Files.delete(Paths.get(file.getPath()));
+    }
+
+    /**
+     * keys are generated internally by this class they're often needed to validate
+     * @return
+     */
+    public String getKey() {
+        return key;
+    }
+
+    /**
+     * name can be generated internally by this class they're often needed to validate
+     * @return
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * file name getter
+     * @return
+     */
+    public String getFileName() {
+        return fileName;
+    }
+
+    /**
+     * app key builder method
+     * @param key
+     * @return
+     */
+    public AppDescriptorDataGen withKey(final String key) {
+        this.key = key;
+        return this;
+    }
+
+    /**
+     * app name builder method
+     * @param name
+     * @return
+     */
+    public AppDescriptorDataGen withName(final String name) {
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * app descriptor builder method
+     * @param description
+     * @return
+     */
+    public AppDescriptorDataGen withDescription(final String description) {
+        this.description = description;
+        return this;
+    }
+
+    /**
+     * file name override  builder method.
+     * @param fileName
+     * @return
+     */
+    public AppDescriptorDataGen withFileName(final String fileName) {
+        this.fileName = fileName;
+        return this;
+    }
+
+    /**
+     * extra params override builder method
+     * @param extraParameters
+     * @return
+     */
+    public AppDescriptorDataGen withExtraParameters(final boolean extraParameters) {
+        this.allowExtraParameters = extraParameters;
+        return this;
+    }
+
+    /**
+     * icon builder method
+     * @param iconUrl
+     * @return
+     */
+    public AppDescriptorDataGen withIconUrl(final String iconUrl) {
+        this.iconUrl = iconUrl;
+        return this;
+    }
+
+    /**
+     * Param builder method
+     * @param name
+     * @param paramDescriptor
+     * @return
+     */
+    public AppDescriptorDataGen param(final String name, final ParamDescriptor paramDescriptor){
+        builder.put(name, paramDescriptor);
+        return this;
+    }
+
+    /**
+     * Param builder method
+     * @param name
+     * @param hidden
+     * @param type
+     * @param label
+     * @param hint
+     * @param required
+     * @return
+     */
+    public AppDescriptorDataGen param(final String name, boolean hidden, final Type type,
+            final String label, final String hint, final boolean required) {
+       return param(name, ParamDescriptor.newParam(StringPool.BLANK, hidden, type, label, hint, required));
+    }
+
+    public AppDescriptorDataGen stringParam(final String name,  boolean hidden, final boolean required) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'hidden' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5358201)